protocol/x/clob/abci.go

```diff
@@ -119,6 +119,10 @@ func PrepareCheckState(
 	keeper *keeper.Keeper,
 	daemonLiquidationInfo *liquidationtypes.DaemonLiquidationInfo,
 ) {
+	// TODO(CORE-538): Is there a cleaner way to check to see if this is genesis?
+	if ctx.BlockHeight() == 0 || ctx.BlockHeight() == 1 {
+		return
+	}
```

Comment:
for my understanding - why do we need to do this?

Reply:
The issue that I ran into is for the first block is that there is no `ProcessProposerMatches` events during `PrepareCheckState`. Cosmos 0.47 didn't run `PrepareCheckState` for the genesis block while Cosmos 0.50 does.

So I keep the original behavior by skipping PrepareCheckState for the genesis block.